### ISSUE TITLE
Rule idea: Warn of using variables without interpolation in css custom properties

### ISSUE THREAD

**Issue by Arkkimaagi** [stylelint-scss/stylelint-scss#597]
I've been bitten a few times by writing SCSS variables in CSS custom properties without interpolation. It would be great if we could lint and warn about this potential issue automatically.

Basically the same problem as here: https://github.com/stylelint-scss/stylelint-scss/issues/388

I read what `dollar-variable-no-missing-interpolation` is supposed to do and I agree with @kristerkari that it's not supposed to fix this issue. So I'm suggesting a new rule to handle this problem.

To re-use the example from that previous issue:

When I accidentally write:
```SCSS
.class {
  $var: 2rem;
  --gap: $var;

  display: grid;
  gap: var(--gap);
}
```
That gets rendered like this and does not work as I intended:
```CSS
.class {
  --gap: $var;

  display: grid;
  gap: var(--gap);
}
```
When I should have written:
```SCSS
.class {
  $var: 2rem;
  --gap: #{$var};

  display: grid;
  gap: var(--gap);
}
```
That gets rendered to:
```CSS
.class {
  --gap: 2rem;

  display: grid;
  gap: var(--gap);
}
```

If you have any other ideas how to tackle this issue with already existing rules, please let me know.

**Comment by nex3** [comment]
It would also be useful to expand this beyond variables to other Sass-specific constructs. In particular, namespaced function calls, known Sass global functions that aren't shadowing CSS globals, and operations outside `calc()` should all get warnings.
